### ISSUE TITLE
Non-limber approximation from N5K

### ISSUE THREAD

**Issue by slosar** [LSSTDESC/CCL#1051]
I somehow surprisingly couldn't find the ticket on non-Limber integrations so I am opening one. Feel free to redirect.
The idea is to implement two winning methods from the N5K challenge.
Here is my proposal:
 
* branch off a new branch off current CCL `main`
 * Add option `nonlimber_integration_method` to `angular_cl`
 * clean up reference to non-limber in the C-layer. Both current non-limber implementation are already python based. So `lib.angular_cl_vec` will now be strictly limber
 *  clean up vestiges on angpow in CCL, it seem it now gone.
 * implement two wining methods in `nonlimber_fkem.py` and `nonlimber_matter.py`.
 * Let `angular_cl` call those and then splice with results from `lib.angular_cl_vec`.
 
Does this make sense?

@damonge @c-d-leonard

**Comment by damonge** [comment]
@slosar yep, that all makes sense. I think it'd be optimal if we were able to do this with a single unified implementation of FFTLog, rather than 3 (these two and the one currently in CCL).

**Comment by damonge** [comment]
(well, I'd also say they shouldn't live in separate files, but within `cells.py`, but those details can be dealt with in review)

**Comment by slosar** [comment]
We wasted a lot of time on FFTlog and then realized that the current CCL already exports CCL C layer fftlog under `fftlog_transform` so we plan to use this one. Paul and Varsha are working on getting that working first (ie. just replacing fftlog with CCL fftlog and working out any 2pi factors)

**Comment by c-d-leonard** [comment]
Sounds good to me. I agree with @damonge that ideally we want a single unified implementation of FFTlog; sounds like that's the plan anyway.
